File: state-machine/src/main/java/br/com/leonardoferreira/poc/statemachine/repository/OrderRepository.java
package br.com.leonardoferreira.poc.statemachine.repository;

import br.com.leonardoferreira.poc.statemachine.domain.Order;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends CrudRepository<Order, Long> {
}

File: spring-events/src/main/kotlin/br/com/leonardoferreira/poc/events/repository/OrderRepository.kt
package br.com.leonardoferreira.poc.events.repository

import br.com.leonardoferreira.poc.events.domain.entity.Order
import br.com.leonardoferreira.poc.events.domain.entity.OrderStatus
import org.springframework.data.jpa.repository.JpaRepository

interface OrderRepository : JpaRepository<Order, Long> {

    fun findByStatusIn(statuses: List<OrderStatus>): List<Order>

}
